File: scalaPath/src/main/java/scalaPath/assistclass/Person.scala
package scalaPath.assistclass

/**
  * Created by liuyong on 2016/3/6.
  */
class Person {
  private var name = ""
  private var age = 0
  def this(name : String){ // 一个辅助构造器
    this() //调用主构造器
    this.name = name
  }
  def this(name: String, age: Int){
    this(name) //調用前一个辅助构造器
    this.age = age
  }
}
object test{
  def main(args: Array[String]) {
    val p1 = new Person //主构造器
    val p2 = new Person("Fred") //辅助构造器
    val p3 = new Person("Fred", 42) //辅助构造器
  }
}

File: SpringLearn/SpringTest/src/main/java/com/spring/bean/Person.java
package com.spring.bean;

import java.util.Date;

/**
 * Created by liuyong on 2016/3/9.
 */
public class Person {
    private String name;

    @Override
    public String toString() {
        return "Person{" +
                "name='" + name + '\'' +
                ", age=" + age +
                ", birthday=" + birthday +
                '}';
    }

    private int age;
    private Date birthday;

    public void init(){
        this.name = "tcliuyong";
        this.age = 12;
        this.birthday = new Date();
    }
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Date getBirthday() {
        return birthday;
    }

    public void setBirthday(Date birthday) {
        this.birthday = birthday;
    }
}
